### ISSUE TITLE
Inconsistent build error messages between dnx451 and dnxcore50

### ISSUE THREAD

**Issue by Tragetaschen** [aspnet/dnx#1829]
In Hosting's Abstractions project, I made the mistake of spelling `CancellationToken` as `CancellatioznToken` (note the `z`) in IApplicationLifetime. When I run `dnu build` in the main project, there are are inconsistent error messages for this situation:

dnx451:
 […]Hosting/src/Microsoft.AspNet.Hosting/ApplicationLifetime.cs(12,40): error CS0738: 'ApplicationLifetime' does not implement interface member 'IApplicationLifetime.ApplicationStarted'. 'ApplicationLifetime.ApplicationStarted' cannot implement 'IApplicationLifetime.ApplicationStarted' because it does not have the matching return type of 'CancellatioznToken'.

That's ok: Roslyn doesn't care at this point that the type itself is invalid and just notifies me that the interface isn't implemented correctly. It least, I'm pointed to the interface.

dnxcore50:
 […]Hosting/src/Microsoft.AspNet.Hosting/ApplicationLifetime.cs(12,40): error CS0246: The type or namespace name 'CancellatioznToken' could not be found (are you missing a using directive or an assembly reference?)

Note, that line 12 is the one with the class declaration also referring to the interface. There is nothing pointing me to the interface and the whole cs file does not contain the offending type name.

**Comment by muratg** [comment]
Is this a Roslyn bug? Should we file it there?

**Comment by davidfowl** [comment]
@Tratcher can you make a simple repro with no external types (just use user defined ones) and try to narrow it down?

**Comment by Tratcher** [comment]
Which version of dnx are you using? I see both errors displayed when compiling against either framework.
Version: 1.0.0-beta5-11706 coreclr x86
```c#
    public interface IFoo
    {
        CancellationToken CancellationToken { get; set; }
    }

    public class Foo : IFoo
    {
        public CancellatioZnToken CancellationToken { get; set; }
    }
```
```
C:\temp\ConsoleApp1\src\ConsoleApp1\Program.cs(23,16): error CS0246: The type or namespace name 'CancellatioZnToken' could not be found (are you missing a using directive or an assembly reference?)
C:\temp\ConsoleApp1\src\ConsoleApp1\Program.cs(21,24): error CS0738: 'Foo' does not implement interface member 'IFoo.CancellationToken'. 'Foo.CancellationToken' cannot implement 'IFoo.CancellationToken' because it does not have the matching return type of 'CancellationToken'.
```
It did not appear to make a difference if I put the interface in a different project.

**Comment by davidfowl** [comment]
I'm going to close this out. It's likely the difference between not having a reference and having a reference to the packages that bring in CancellatonToken. Other than that, it could only be a roslyn bug.

**Comment by Tragetaschen** [comment]
I failed to create a repro from scratch as well, so I started taking away stuff from Hosting while keeping the error intact. The result is [here](https://github.com/Tragetaschen/Hosting/tree/weird-error-message) in my Hosting fork (I promise, I won't PR this).

As the tree is setup now, you `dnu restore` the projects and then `dnu build` in `Microsoft.AspNet.Hosting`.

The important bit is, that that main project pulls in CoreCLR's `System.Threading` indirectly using the reference to `Microsoft.Framework.Runtime.Abstractions`. If you remove that (otherwise unused) reference and add one in dnxcore50 to `System.Threading`, the error is "correct" for CoreCLR.
